commands/livecmd.go

```diff
@@ -79,8 +80,11 @@ func newFactory(cmd *cobra.Command, version string) cluster.Factory {
 	flags := cmd.PersistentFlags()
 	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
 	kubeConfigFlags.AddFlags(flags)
+	cachingConfigFlags := &factory.CachingRESTClientGetter{
+		Delegate: kubeConfigFlags,
+	}
 	userAgentKubeConfigFlags := &cfgflags.UserAgentKubeConfigFlags{
-		Delegate:  kubeConfigFlags,
+		Delegate:  cachingConfigFlags,
```

Comment:
I was trying to understand the description of the PR with the code change, but couldn't decipher :)
I could see layers of flags here, but will be nice to have a brief comment explaining what is happening here.

Reply:
Added a comment here to explain why we are using the `CachingRESTClientGetter`.